browser/brave_content_browser_client.cc

```diff
@@ -198,6 +213,39 @@ void BindCosmeticFiltersResources(
       std::move(receiver));
 }
 
+#if BUILDFLAG(BRAVE_ADS_ENABLED)
+void BindBraveAdsDriver(
+    content::RenderFrameHost* const frame_host,
+    mojo::PendingReceiver<brave_ads::mojom::BraveAdsDriver> receiver) {
+  auto* context = frame_host->GetBrowserContext();
+  auto* profile = Profile::FromBrowserContext(context);
+  if (brave::IsRegularProfile(profile)) {
+    content::WebContents* web_contents =
+        content::WebContents::FromRenderFrameHost(frame_host);
+    brave_ads::AdsService* ads_service =
```

Comment:
you need null checks for these

Reply:
There are null checks for `ads_service` and `rewards_service` in `brave_ads::BraveAdsDriver` already.
But it is a good idea to check it for nulls here and create `brave_ads::BraveAdsDriverPrivate` if null

Reply:
Moved checks and factories to BatAdsHost